test/forms/common/DateSpec.scala

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.common
+
+import forms.common.Date._
+import org.scalatest.{MustMatchers, WordSpec}
+import play.api.data.FormError
+import play.api.libs.json.{JsObject, JsString, JsValue}
+
+class DateSpec extends WordSpec with MustMatchers {
```

Comment:
I cannot agree. Scalacheck is randomized to the extent, that it makes tests unstable (failing because of scalacheck problems, not implementation). This is why I am not using it in my tests.
I would consider it more appropriate to test different permutations and combinations in Acceptance Tests, or create another group of tests just for this purpose. Or ideally, use a tool, which is not introducing instabilities, but I am not aware of any at the moment.

Reply:
if they are failing it would be failing because we are missing some permutations , I agree not using in Controller but at for level we should use this as it is used extensively
on MDTP

Reply:
I'll look into that for the next PR. These are tests only and applying scalacheck is form of refactoring.